src/Shared/FileUtilitiesRegex.cs

```diff
@@ -19,13 +15,21 @@ namespace Microsoft.Build.Shared
     /// </summary>
     internal static class FileUtilitiesRegex
     {
-        // regular expression used to match file-specs beginning with "<drive letter>:" 
-        internal static readonly Regex DrivePattern = new Regex(@"^[A-Za-z]:", RegexOptions.Compiled);
+        // regular expression used to match file-specs comprising exactly "<drive letter>:" (with no trailing characters)
+        internal static readonly Regex DrivePattern = new Regex(@"^[A-Za-z]:$", RegexOptions.Compiled);
```

Comment:
Since this is matching both the beginning and end of the string, wouldn't ```DrivePattern.IsMatch()``` now always return false for any string that isn't exactly ```[drive letter]:``` such as ```C:\foo```? That's very different from the original behavior which only checks the prefix for a match.

Reply:
The change was intentional&mdash;the usage in `FileUtilities` needed an exact match.

I made a mistake wrt the other usage in `Modifiers.GetItemSpecModifier`. That usage _does_ need starts-with logic. I've pushed a fix.